Add note about pre-migration for Pulp Container content (#3052)
theforeman/foreman-documentation: guides/common/modules/con_performing-post-upgrade-tasks.adoc

[id="performing-post-upgrade-tasks_{context}"]
= Performing post-upgrade tasks

* Optional: If the default provisioning templates have been changed during the upgrade, recreate any templates cloned from the default templates.
If the custom code is executed before and/or after the provisioning process, use custom provisioning snippets to avoid recreating cloned templates.
For more information about configuring custom provisioning snippets, see {ProvisioningDocURL}Creating_Custom_Provisioning_Snippets_provisioning[Creating Custom Provisioning Snippets] in _{ProvisioningDocTitle}_.

> 

Reviewer: ```suggestion
ifdef::katello,orcharhino,satellite[]
```

* Pulp is introducing more data about container manifests to the API.
This information allows Katello to display manifest labels, annotations, and information about the manifest type, such as if it is bootable or represents flatpak content.
As a result, migrations must be performed to pull this content from manifests into the database.

> 

Reviewer: ```suggestion
endif::[]
```

> ifdef::katello[]

Reviewer: ```suggestion
ifdef::katello[]
+
```

This migration takes time, so if you depend on container content and need minimal upgrade downtime, run the following command after upgrading for a pre-migration.

> 

Reviewer: ```suggestion
+
```

[options="nowrap" subs="attributes"]
----
# foreman-maintain advanced procedure run pulpcore-container-handle-image-metadata
----
This command migrates data in the background without any need for downtime.
endif::[]

> 

Reviewer: ```suggestion
```

> ifdef::satellite[]

Reviewer: ```suggestion
ifdef::satellite[]
+
```

[id="performing-post-upgrade-tasks_{context}"]
= Performing post-upgrade tasks

* Optional: If the default provisioning templates have been changed during the upgrade, recreate any templates cloned from the default templates.
If the custom code is executed before and/or after the provisioning process, use custom provisioning snippets to avoid recreating cloned templates.
For more information about configuring custom provisioning snippets, see {ProvisioningDocURL}Creating_Custom_Provisioning_Snippets_provisioning[Creating Custom Provisioning Snippets] in _{ProvisioningDocTitle}_.
ifdef::katello,orcharhino,satellite[]
* Pulp is introducing more data about container manifests to the API.
This information allows Katello to display manifest labels, annotations, and information about the manifest type, such as if it is bootable or represents flatpak content.

> As a result, migrations must be performed to pull this content from manifests into the database.

Reviewer: ```suggestion
As a result, migration must be performed to pull this content from container manifests into the database.
```
I'm wondering why the plural of migrations - is there a reason? Also, I'd prefer to be specific about which manifests are in question so that the user doesn't confuse them with subscription manifests.

endif::[]

> ifdef::katello,orcharhino[]

Reviewer 2: ```suggestion
ifdef::katello[]
```

+

> This migration takes time, so if you depend on container content and need minimal upgrade downtime, run the following command after upgrading for a pre-migration.

Reviewer: ```suggestion
This migration takes time, so if you depend on container content and need minimal upgrade downtime, run the following command for a pre-migration after upgrading:
```
This order of the phrases looks cleaner to me. Plus a colon, because the sentence introduces a block code.

> This migration takes time, so if you depend on container content and need minimal upgrade downtime, run the following command for a pre-migration after upgrading:

Reviewer: One more thing; we probably don't have to say "after upgrading" since the context are post-upgrade tasks. And we should clarify where the command is to be run.

```suggestion
This migration takes time, so if you depend on container content and need minimal upgrade downtime, run the following command on {ProjectServer} for a pre-migration:
```

+
[options="nowrap" subs="attributes"]
----
# foreman-maintain advanced procedure run pulpcore-container-handle-image-metadata
----

> This command migrates data in the background without any need for downtime.

Reviewer: ```suggestion
+
This command migrates data in the background without any need for downtime.
```
For clean formatting.
